src/contexts/filter-context/filter-reducer.jsx

```diff
@@ -0,0 +1,30 @@
+export const filterReducer = (state, action) => {
+  switch (action.type) {
+    case "SET_PRIORITY":
+      return { ...state, priority: action.payload, isFilterApplied: true };
+    case "SET_SEARCH":
+      return {
+        ...state,
+        searchText: action.payload,
+        isFilterApplied:
+          action.payload === "" ? (state.isFilterApplied ? false : true) : true,
+      };
+    case "SET_DATE":
+      return { ...state, sortByDate: action.payload, isFilterApplied: true };
+    case "SET_TAG":
+      const newTag = state.tags.includes(action.payload)
+        ? state.tags.filter(note => note !== action.payload)
+        : [...state.tags, action.payload];
+      return { ...state, tags: newTag, isFilterApplied: true };
+    case "CLEAR_FILTER":
+      return {
+        sortByDate: "",
+        priority: "",
+        tags: [],
+        isFilterApplied: false,
+        searchText: "",
+      };
```

Comment:
You can define this as an initial state constant and call that whenever you want to reset it

Reply:
Oh yes, thanks for this, code looks cleaner now.